### PR TITLE
[KOGITO-8644] - Review "applied" status attribute

### PR DESCRIPTION
**Pull request by desmax74**
See: https://issues.redhat.com/browse/KOGITO-8644

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by radtriste — APPROVED** [review]

File: .github/workflows/check-container-builder.yml [hunk 1]
```diff
@@ -62,8 +62,8 @@ jobs:
         run: |
           wget http://security.ubuntu.com/ubuntu/pool/main/s/shadow/libsubid4_4.11.1+dfsg1-2ubuntu1.1_amd64.deb
           sudo dpkg -i libsubid4_4.11.1+dfsg1-2ubuntu1.1_amd64.deb
-          wget http://ftp.us.debian.org/debian/pool/main/libp/libpod/podman_4.3.1+ds1-6+b2_amd64.deb
-          sudo dpkg -i podman_4.3.1+ds1-6+b2_amd64.deb
+          wget http://ftp.us.debian.org/debian/pool/main/libp/libpod/podman_4.4.0+ds1-1_amd64.deb
```
Comment (spolti) [inline .github/workflows/check-container-builder.yml:65]: there is another PR with this change.

Comment (spolti) [reply .github/workflows/check-container-builder.yml:65]: can we wait the @radtriste's PR to get merged?

Comment (radtriste) [reply .github/workflows/check-container-builder.yml:65]: @spolti it does care much which PR will be merged first. changes are the same. rebase will handle that ;)

**Review by spolti — COMMENTED** [review]
@desmax74 please make sure to add some unit tests.

**Review by ricardozanini — APPROVED** [review]

File: controllers/profiles/reconciler_prod_test.go [hunk 2]
```diff
@@ -129,3 +130,34 @@ func Test_deployWorkflowReconciliationHandler_handleObjects(t *testing.T) {
...
+	handler := &deployWorkflowReconciliationState{
+		stateSupport: fakeReconcilerSupport(client),
+		ensurers:     newProdObjectEnsurers(&stateSupport{logger: &logger, client: client}),
+	}
+	result, objects, err := handler.Do(context.TODO(), workflow)
+	assert.Greater(t, result.RequeueAfter, int64(0))
+	assert.NoError(t, err)
+	assert.NotNil(t, result)
+	assert.Len(t, objects, 2)
+	// then we load a workflow with metadata.generation set to 1
```
Comment (spolti) [inline controllers/profiles/reconciler_prod_test.go:151]: instead update the generation, imho, would be better to compare if the content of the workflow has changed.

Comment (desmax74) [reply controllers/profiles/reconciler_prod_test.go:151]: I can add the compare the content of the workflow, but is the metadata.generation value compared to the status.observedGeneration the kickstart for the new build in this new code

**Comment by ricardozanini** [comment]
Merging to unlock Kris. I'll add a few items to this feature in my upcoming PR.

**Pull request merged**